### ISSUE TITLE
Allow re-exports of declarations without having to export `module Whatever`

### ISSUE THREAD

**Issue by garyb** [purescript/purescript#1902]
_Requiring_ the use of `module` to re-export members is probably a bit weird, it seems like we should allow this as well:

``` purescript
module A where

  foo :: String
  foo = "hello"

module B (foo, bar) where

  import A (foo)

  bar :: String
  bar = "world"

module Main where

  import Control.Monad.Eff.Console (log)

  main = log (B.foo <> B.bar)
```

**Comment by paf31** [comment]
This seems harmless enough. I'm just wary of adding more features to the module/namespace system, if we're planning any radical changes before 0.9.

**Comment by garyb** [comment]
Yeah, it's not something I intended to work on right now, just thought I'd throw it on the pile. It's functionally identical to something we can already do using `module` re-exports with a pseudo-module, so it's just a convenience really.

**Comment by garyb** [comment]
I'm also fed up of working on module/name stuff and would like to go back to some of the other stuff on my list :smile:

**Comment by timjs** [comment]
Stumbled upon this problem today. Any ideas about supporting it?

**Comment by semyon2105** [comment]
This would be nice to have. I hit a case where I needed to re-export only select imports of another module. `module X` exports all of them.
